test/e2e/user_management_test.go

```diff
@@ -42,7 +45,99 @@ func (s *userManagementTestSuite) TearDownTest() {
+
+	// Deactivate the user
+	userSignup.Spec.Deactivated = true
+	err := s.awaitility.Client.Update(context.TODO(), userSignup)
+	require.NoError(s.T(), err)
+	s.T().Logf("user signup '%s' set to deactivated", userSignup.Name)
+
+	err = s.hostAwait.WaitUntilMasterUserRecordDeleted(mur.Name)
+	require.NoError(s.T(), err)
+}
+
+func (s *userManagementTestSuite) TestUserBanning() {
+
+	s.T().Run("user banning", func(t *testing.T) {
+		// when
+		s.setApprovalPolicyConfig("automatic")
+
+		// then
+		s.checkUserBanned()
+	})
+
+}
+
+func (s *userManagementTestSuite) checkUserBanned() {
+	s.T().Run("ban provisioned usersignup", func(t *testing.T) {
+		s.setApprovalPolicyConfig("automatic")
+
+		// Create a new UserSignup and confirm it was approved automatically
+		userSignup, _ := s.createUserSignupAndAssertAutoApproval(false)
+
+		// Create the BannedUser
+		s.createAndCheckBannedUser(userSignup.Annotations[v1alpha1.UserSignupUserEmailAnnotationKey])
+
+		// Confirm the user is banned
+		_, err := s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second*10)).WaitForUserSignup(userSignup.Name,
+			wait.UntilUserSignupHasConditions(approvedAutomaticallyAndBanned()...))
+		require.NoError(s.T(), err)
+
+		// Confirm that a MasterUserRecord is deleted
+		_, err = s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second * 10)).WaitForMasterUserRecord(userSignup.Spec.Username)
+		require.Error(s.T(), err)
+	})
+
+	s.T().Run("create usersignup with preexisting banneduser", func(t *testing.T) {
+		s.setApprovalPolicyConfig("automatic")
+
+		id := uuid.NewV4().String()
+		email := "testuser" + id + "@test.com"
+		s.createAndCheckBannedUser(email)
+
+		_ = s.createAndCheckUserSignupNoMUR(false, "testuser"+id, email, banned()...)
+	})
+
+	s.T().Run("register new user with preexisting ban", func(t *testing.T) {
+		s.setApprovalPolicyConfig("automatic")
+
+		id := uuid.NewV4().String()
+		email := "testuser" + id + "@test.com"
+		s.createAndCheckBannedUser(email)
+
+		// Get valid generated token for e2e tests. IAT claim is overriden
+		// to avoid token used before issued error.
+		identity0 := authsupport.NewIdentity()
+		emailClaim0 := authsupport.WithEmailClaim(email)
+		token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0)
+		require.NoError(s.T(), err)
+
+		route := s.awaitility.RegistrationServiceURL
+
+		// Call signup endpoint with a valid token to initiate a signup process
+		req, err := http.NewRequest("POST", route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("Authorization", "Bearer "+token0)
+		req.Header.Set("content-type", "application/json")
+
+		resp, err := httpClient.Do(req)
+		require.NoError(s.T(), err)
+		defer close(s.T(), resp)
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+		assert.Equal(s.T(), http.StatusInternalServerError, resp.StatusCode)
```

Comment:
Is it also possible to check the content of the error?

Reply:
Done